recipes/imagemgagick/all/conanfile.py

```diff
@@ -0,0 +1,349 @@
+
+    _source_subfolder = "ImageMagick"  # name is important, VisualMagick uses relative paths to it
+    _build_subfolder = "build_subfolder"
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os == 'Windows' and self.settings.compiler == 'gcc' and os.name == 'nt'
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == 'Visual Studio'
+
+    @property
+    def _modules(self):
+        return ['Magick++', 'MagickWand', 'MagickCore']
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires('pango/1.48.5')
+        if self.options.zlib:
+            self.requires('zlib/1.2.11')
+        if self.options.bzlib:
+            self.requires('bzip2/1.0.8')
+        if self.options.lzma:
+            self.requires('xz_utils/5.2.5')
+        if self.options.lcms:
+            self.requires('lcms/2.11')
+        if self.options.openexr:
+            self.requires('openexr/2.5.5')
+        if self.options.jpeg:
+            self.requires('libjpeg/9d')
+        if self.options.openjp2:
+            self.requires('openjpeg/2.4.0')
+        if self.options.png:
+            self.requires('libpng/1.6.37')
+        if self.options.tiff:
+            self.requires('libtiff/4.2.0')
+        if self.options.webp:
+            self.requires('libwebp/1.2.0')
+        if self.options.xml:
+            self.requires('libxml2/2.9.10')
+        if self.options.freetype:
+            self.requires('freetype/2.10.4')
+
+    def source(self):
+        tools.get("{0}/archive/{1}.tar.gz".format(self.homepage, self.version))
+        extracted_dir = "ImageMagick-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+        tools.get('https://github.com/ImageMagick/VisualMagick/archive/refs/heads/main.zip')
+        os.rename('VisualMagick-main', 'VisualMagick')
+
+    def _copy_pkg_config(self, name):
+        if name not in self.deps_cpp_info.deps:
+            return
+        root = self.deps_cpp_info[name].rootpath
+        pc_dir = os.path.join(root, 'lib', 'pkgconfig')
+        pc_files = glob.glob('%s/*.pc' % pc_dir)
+        if not pc_files:
+            pc_files = glob.glob('%s/*.pc' % root)
+        for pc_name in pc_files:
+            new_pc = os.path.join('pkgconfig', os.path.basename(pc_name))
+            self.output.info('copying .pc file %s' % os.path.basename(pc_name))
+            shutil.copy(pc_name, new_pc)
+            prefix = tools.unix_path(root) if os.name == 'nt' else root
+            tools.replace_prefix_in_pc_file(new_pc, prefix)
+
+    def build(self):
+        if self._is_msvc:
+            self._build_msvc()
+        else:
+            self._build_configure()
+
+    def _build_msvc(self):
+        # remove unnecessary dependencies from config
+        for lib in ['bzlib', 'glib', 'lcms', 'libxml', 'lqr', 'ttf', 'zlib']:
```

Comment:
I think the `tools.replace_in_file` calls in this method can be extracted to a patch.
This would reduce the size of the recipe a lot.

Reply:
I replaced all of these that wren't conditional with a patch.